src/twin/system_info.rs

```diff
@@ -4,13 +4,64 @@ use crate::system;
 use anyhow::{bail, Context, Result};
 use async_trait::async_trait;
 use azure_iot_sdk::client::{IotHubClient, IotMessage};
+use futures::StreamExt;
 use lazy_static::lazy_static;
 use log::{debug, warn};
 use serde::Serialize;
 use serde_json::json;
 use std::env;
 use std::path::Path;
+use sysinfo::Disks;
+use sysinfo::{Components, CpuRefreshKind, MemoryRefreshKind, RefreshKind, System};
+use time::format_description::well_known::Rfc3339;
 use tokio::sync::mpsc;
+use tokio::time::interval;
+use tokio::time::Duration;
+
+lazy_static! {
+    static ref REFRESH_SYSTEM_INFO_INTERVAL_SECS: u64 = {
+        const REFRESH_SYSTEM_INFO_INTERVAL_SECS_DEFAULT: &str = "60";
+        std::env::var("REFRESH_SYSTEM_INFO_INTERVAL_SECS")
+            .unwrap_or(REFRESH_SYSTEM_INFO_INTERVAL_SECS_DEFAULT.to_string())
+            .parse::<u64>()
+            .expect("cannot parse REFRESH_SYSTEM_INFO_INTERVAL_SECS env var")
+    };
+}
+
+#[derive(Serialize)]
+struct Label {
+    edge_device: String,
+    module_name: String,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    sensor: Option<String>,
+}
+
+#[derive(Serialize)]
+struct Metric {
+    #[serde(rename = "TimeGeneratedUtc")]
+    time_generated_utc: String,
+    #[serde(rename = "Name")]
+    name: String,
+    #[serde(rename = "Value")]
+    value: f64,
+    #[serde(rename = "Labels")]
+    labels: Label,
+}
+
+impl Metric {
+    fn new(time: String, name: &str, value: f64, device: String, sensor: Option<String>) -> Metric {
```

Comment:
Is there a reason for only "name" to be `&str`?

Reply:
no, will change to String.

Reply:
done